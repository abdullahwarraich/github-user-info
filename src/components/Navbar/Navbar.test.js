import { render, fireEvent } from '@testing-library/react'
import NavMenu from './index'

const mockHistoryPush = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  })
}))

describe('Navbar', () => {
  let navMenuRender

  afterEach(() => {
    navMenuRender.unmount()
  })

  test('should render Navbar', async () => {
    navMenuRender = await render(<NavMenu />)

    const profileLink = await navMenuRender.findByText('Profile')
    fireEvent.click(profileLink)
    expect(mockHistoryPush).toHaveBeenCalledWith('/profile')

    const gistLink = await navMenuRender.findByText('Gists')
    fireEvent.click(gistLink)
    expect(mockHistoryPush).toHaveBeenCalledWith('/gists')

    const orgsLink = await navMenuRender.findByText('Orgs')
    fireEvent.click(orgsLink)
    expect(mockHistoryPush).toHaveBeenCalledWith('/orgs')

     expect(navMenuRender).toMatchSnapshot()
  })
})
