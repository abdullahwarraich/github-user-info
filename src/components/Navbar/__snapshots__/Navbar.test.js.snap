// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-eCImvq ghfczK"
      >
        <nav
          class="sc-bdvvaa iwfTtt"
        >
          <div
            class="sc-gsDJrp gPfrBN"
          >
            <div
              class="leftMenu"
            >
              <ul
                class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
                role="menu"
              >
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="padding-left: 24px;"
                >
                  Profile
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="padding-left: 24px;"
                >
                  Gists
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
                  role="menuitem"
                  style="padding-left: 24px;"
                >
                  Orgs
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  role="menuitem"
                  style="padding-left: 24px;"
                >
                  Search New User
                </li>
              </ul>
            </div>
            <button
              class="ant-btn ant-btn-primary barsMenu"
              type="button"
            >
              <span
                class="sc-hKwCoD jeKmDZ"
              />
            </button>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-eCImvq ghfczK"
    >
      <nav
        class="sc-bdvvaa iwfTtt"
      >
        <div
          class="sc-gsDJrp gPfrBN"
        >
          <div
            class="leftMenu"
          >
            <ul
              class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
              role="menu"
            >
              <li
                class="ant-menu-item ant-menu-item-only-child"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Profile
              </li>
              <li
                class="ant-menu-item ant-menu-item-only-child"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Gists
              </li>
              <li
                class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Orgs
              </li>
              <li
                class="ant-menu-item ant-menu-item-only-child"
                role="menuitem"
                style="padding-left: 24px;"
              >
                Search New User
              </li>
            </ul>
          </div>
          <button
            class="ant-btn ant-btn-primary barsMenu"
            type="button"
          >
            <span
              class="sc-hKwCoD jeKmDZ"
            />
          </button>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
